### PR TITLE
Set default values for configs

### LINKED ISSUE #947: Refactor configurations
### Describe your problem(s)

Refactor micro-gw.conf file and add default conf file 
Default config values should be set to constants.

### How will you implement it

---

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Issues

#947

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-cli/src/main/resources/templates/jwtRevocation.mustache [hunk 1]
```diff
@@ -2,7 +2,7 @@ map<string> receivedRevokedTokenMap = gateway:getRevokedTokenMap();
     boolean jmsListenerStarted = gateway:initiateTokenRevocationJmsListener();
 
     boolean useDefault = gateway:getConfigBooleanValue(gateway:PERSISTENT_MESSAGE_INSTANCE_ID,
-    gateway:PERSISTENT_USE_DEFAULT, false);
+        gateway:PERSISTENT_USE_DEFAULT, gateway:DEFAULT_PERSISTENT_USE_DEFAULT);
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/jwtRevocation.mustache]: indentation issue

Comment (AmaliMatharaarachchi) [reply components/micro-gateway-cli/src/main/resources/templates/jwtRevocation.mustache]: I believe the indentation is correct here. Kindly insert the suggestion<cmd-g> if not.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/constants/constants.bal [hunk 2]
```diff
@@ -262,14 +261,9 @@ public const string JMS_CONNECTION_PASSWORD = "jmsConnectionPassword";
 public const string THROTTLE_ENDPOINT_URL = "throttleEndpointUrl";
 public const string THROTTLE_ENDPOINT_BASE64_HEADER = "throttleEndpointbase64Header";
 
-public const string TOKEN_REVOCATION_CONF_INSTANCE_ID = "tokenRevocationConfig";
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/constants/constants.bal]: I hope these are unused or duplicate constants

Comment (AmaliMatharaarachchi) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/constants/constants.bal]: Most of them are duplicated. So I removed them.

File: distribution/resources/conf/micro-gw-default.conf [hunk 3]
```diff
@@ -0,0 +1,141 @@
+[listenerConfig]
```
[the remark this answers is not in the document]
Comment (AmaliMatharaarachchi) [reply distribution/resources/conf/micro-gw-default.conf]: +1

File: components/micro-gateway-core/src/main/ballerina/src/gateway/constants/micro_gw_conf_defaults.bal [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+// under the License.
+
+// default should bind to 0.0.0.0, not localhost. Else will not work in dockerized environments.
+public const string DEFAULT_CONF_HOST = "0.0.0.0";
+public const int DEFAULT_HTTP_PORT = 9090;
+public const int DEFAULT_HTTPS_PORT = 9095;
+public const string DEFAULT_KEY_STORE_PATH = "${ballerina.home}/bre/security/ballerinaKeystore.p12";
+public const string DEFAULT_KEY_STORE_PASSWORD = "ballerina";
+public const string DEFAULT_TRUST_STORE_PATH = "${ballerina.home}/bre/security/ballerinaTruststore.p12";
+public const string DEFAULT_TRUST_STORE_PASSWORD = "ballerina";
```
[the remark this answers is not in the document]
Comment (AmaliMatharaarachchi) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/constants/micro_gw_conf_defaults.bal]: Kindly consider that Ballerina does not allow us to assign a const value to another const value.

**Comment by AmaliMatharaarachchi** [comment]
@praminda Thanks. Fixed as review comments in https://github.com/wso2/product-microgateway/pull/948/commits/5c7321c1a470393cb8484ff7a273b2825bf1ed0b

**Pull request merged**